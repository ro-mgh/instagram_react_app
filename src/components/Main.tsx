import React, { useState, useEffect, FunctionComponent } from "react";
import PhoneAnimation from "./PhoneAnimation";

const Main: FunctionComponent = () => {
  const userId = 123;
  if (userId) {
    return (
      <article className="article-sign">
        <div className="div-sign-phone">
          <PhoneAnimation />
        </div>
        <div className="div-sign-form">SigninComponent</div>
      </article>
    );
  }
  return <div>i am here</div>;
};

export default Main;
