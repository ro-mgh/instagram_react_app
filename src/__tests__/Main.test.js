import { unmountComponentAtNode } from "react-dom";
import React from "react";
import { render, screen } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

import Main from "../components/Main";

import configureStore from "redux-mock-store";

import { Provider } from "react-redux";
import renderer from "react-test-renderer";

const mockStore = configureStore([]);

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("rendering Main component", () => {
  let store;
  let component;
  const history = createMemoryHistory();

  // beforeEach(() => {});

  it("renders Main component without props", () => {
    store = mockStore({
      firebaseReducer: {
        auth: "tbc",
      },
    });

    component = render(
      <Provider store={store}>
        <Router history={history}>
          <Main />
        </Router>
      </Provider>
    );
    // screen.debug();
    expect(component.container).toMatchSnapshot();
    // expect(component.container.firstChild).
  });

  // it("renders Main component with user", () => {
  //   expect(component.toJSON()).toMatchSnapshot();
  // });
});
