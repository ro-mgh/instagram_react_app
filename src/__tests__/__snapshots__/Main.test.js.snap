// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Main component renders Main component with no user auth, and redirecting to signin 1`] = `
<div>
  <div>
    <div>
      <article
        class="article-sign"
      >
        <div
          class="div-sign-phone"
        >
          <div
            class="div-sign-phone-top"
          >
            <img
              alt=""
              class="div-sign-photo-img"
              src="phone_signup_carousel_1.jpg"
            />
          </div>
        </div>
        <div
          class="div-sign-form"
        >
          <div
            class="sign-mainfield"
          >
            <div
              class="customlogo-wrapper-sign"
            >
              <h1
                class="sign-logo"
              >
                Insta
              </h1>
              <div
                class="customlogo-div-sign"
              >
                 | by RM
              </div>
            </div>
            <div
              data-testid="signin"
            >
              <div
                class="signin-demouser-wrapper"
              >
                <button
                  class="sign-up-loginFacebook signin-demouser"
                >
                  Sign in as demouser
                </button>
              </div>
              <div
                class="sign-form-div"
              >
                <form
                  class="sign-form"
                  novalidate=""
                >
                  <div
                    class="sign-input-field"
                  >
                    <label
                      class="sign-input-label"
                    >
                      <span
                        class="sign-input-span"
                      >
                        Phone number, username, or email
                      </span>
                      <input
                        aria-label="Phone number, username, or email"
                        aria-required="true"
                        autocapitalize="off"
                        autocorrect="off"
                        class="sign-input-form"
                        maxlength="75"
                        name="username"
                        type="text"
                        value=""
                      />
                    </label>
                  </div>
                  <div
                    class="sign-input-field"
                  >
                    <label
                      class="sign-input-label"
                    >
                      <span
                        class="sign-input-span"
                      >
                        Password
                      </span>
                      <input
                        aria-label="Password"
                        aria-required="true"
                        autocapitalize="off"
                        autocorrect="off"
                        class="sign-input-form"
                        name="password"
                        type="password"
                        value=""
                      />
                    </label>
                    <div />
                  </div>
                  <div
                    class="sign-button-div"
                  >
                    <button
                      class="sign-button-submit"
                    >
                      Log in
                    </button>
                  </div>
                </form>
              </div>
            </div>
            <div
              class="sign-or"
            >
              <div
                class="sign-or-dash"
              />
              <div
                class="sign-or-text"
              >
                or
              </div>
              <div
                class="sign-or-dash"
              />
            </div>
            <button
              class="sign-loginFacebook"
            >
              <span
                class="sign-loginFacebook-logo"
              />
              <span
                class="sign-loginFacebook-text"
              >
                Log in with Facebook
              </span>
            </button>
            <p
              class="sign-forgot-password"
            >
              Forgot password?
            </p>
          </div>
          <div
            class="sign-redirect"
          >
            <div
              class="sign-redirect-p"
            >
              Don't have an account?
               
              <a
                href="/signup"
              >
                <div
                  class="sign-redirect-a"
                >
                  <span
                    class="sign-redirect-span"
                  >
                    Sign up
                  </span>
                </div>
              </a>
            </div>
          </div>
          <p
            class="sign-links-p"
          >
            Get the app.
          </p>
          <div
            class="sign-links-div"
          >
            <img
              alt=""
              class="sign-links-img"
              src="link_signup_appstore.png"
            />
            <img
              alt=""
              class="sign-links-img"
              src="link_signup_googlestore.png"
            />
          </div>
        </div>
      </article>
      <footer>
        <div
          class="footer-div-main"
        >
          <div
            class="footer-div-top"
          >
            <div
              class="footer-div-top-placeholder"
            >
              <a
                class="footer-text"
                href="https://about.instagram.com/"
                rel="nofollow noopener noreferrer"
                target="_blank"
              >
                About
              </a>
              <a
                class="footer-text"
                href="https://about.instagram.com/blog/"
                rel="nofollow noopener noreferrer"
                target="_blank"
              >
                Blog
              </a>
              <a
                class="footer-text"
                href="/about/jobs/"
              >
                Jobs
              </a>
              <a
                class="footer-text"
                href="https://help.instagram.com/"
                rel="nofollow noopener noreferrer"
                target="_blank"
              >
                Help
              </a>
              <a
                class="footer-text"
                href="/developer/"
              >
                API
              </a>
              <a
                class="footer-text"
                href="/legal/privacy/"
              >
                Privacy
              </a>
              <a
                class="footer-text"
                href="/legal/terms/"
              >
                Terms
              </a>
              <a
                class="footer-text"
                href="/directory/profiles/"
              >
                Top Accounts
              </a>
              <a
                class="footer-text"
                href="/directory/hashtags/"
              >
                Hashtags
              </a>
              <a
                class="footer-text"
                href="/explore/locations/"
              >
                Locations
              </a>
            </div>
            <div
              class="footer-div-bottom-placeholder"
            >
              <a
                class="footer-text"
                href="/topics/beauty/"
              >
                Beauty
              </a>
              <a
                class="footer-text"
                href="/topics/dance-and-performance/"
              >
                Dance & Performance
              </a>
              <a
                class="footer-text"
                href="/topics/fitness/"
              >
                Fitness
              </a>
              <a
                class="footer-text"
                href="/topics/food-and-drink/"
              >
                Food & Drink
              </a>
              <a
                class="footer-text"
                href="/topics/home-and-garden/"
              >
                Home & Garden
              </a>
              <a
                class="footer-text"
                href="/topics/music/"
              >
                Music
              </a>
              <a
                class="footer-text"
                href="/topics/visual-arts/"
              >
                Visual Arts
              </a>
            </div>
          </div>
          <div
            class="footer-div-bottom"
          >
            <div
              class="footer-text"
            >
              English
            </div>
            <div
              class="footer-text"
            >
              Â© 2021 Instagram by RM
            </div>
          </div>
        </div>
      </footer>
    </div>
  </div>
</div>
`;

exports[`rendering Main component renders Main component without props 1`] = `
<div>
  <div>
    <div
      data-testid="loader"
    >
      <span>
        <svg
          class="loader"
          height="50"
          viewBox="0 0 50 50"
          width="50"
        >
          <path
            d="M25 1c-6.52 0-7.34.03-9.9.14-2.55.12-4.3.53-5.82 1.12a11.76 11.76 0 0 0-4.25 2.77 11.76 11.76 0 0 0-2.77 4.25c-.6 1.52-1 3.27-1.12 5.82C1.03 17.66 1 18.48 1 25c0 6.5.03 7.33.14 9.88.12 2.56.53 4.3 1.12 5.83a11.76 11.76 0 0 0 2.77 4.25 11.76 11.76 0 0 0 4.25 2.77c1.52.59 3.27 1 5.82 1.11 2.56.12 3.38.14 9.9.14 6.5 0 7.33-.02 9.88-.14 2.56-.12 4.3-.52 5.83-1.11a11.76 11.76 0 0 0 4.25-2.77 11.76 11.76 0 0 0 2.77-4.25c.59-1.53 1-3.27 1.11-5.83.12-2.55.14-3.37.14-9.89 0-6.51-.02-7.33-.14-9.89-.12-2.55-.52-4.3-1.11-5.82a11.76 11.76 0 0 0-2.77-4.25 11.76 11.76 0 0 0-4.25-2.77c-1.53-.6-3.27-1-5.83-1.12A170.2 170.2 0 0 0 25 1zm0 4.32c6.4 0 7.16.03 9.69.14 2.34.11 3.6.5 4.45.83 1.12.43 1.92.95 2.76 1.8a7.43 7.43 0 0 1 1.8 2.75c.32.85.72 2.12.82 4.46.12 2.53.14 3.29.14 9.7 0 6.4-.02 7.16-.14 9.69-.1 2.34-.5 3.6-.82 4.45a7.43 7.43 0 0 1-1.8 2.76 7.43 7.43 0 0 1-2.76 1.8c-.84.32-2.11.72-4.45.82-2.53.12-3.3.14-9.7.14-6.4 0-7.16-.02-9.7-.14-2.33-.1-3.6-.5-4.45-.82a7.43 7.43 0 0 1-2.76-1.8 7.43 7.43 0 0 1-1.8-2.76c-.32-.84-.71-2.11-.82-4.45a166.5 166.5 0 0 1-.14-9.7c0-6.4.03-7.16.14-9.7.11-2.33.5-3.6.83-4.45a7.43 7.43 0 0 1 1.8-2.76 7.43 7.43 0 0 1 2.75-1.8c.85-.32 2.12-.71 4.46-.82 2.53-.11 3.29-.14 9.7-.14zm0 7.35a12.32 12.32 0 1 0 0 24.64 12.32 12.32 0 0 0 0-24.64zM25 33a8 8 0 1 1 0-16 8 8 0 0 1 0 16zm15.68-20.8a2.88 2.88 0 1 0-5.76 0 2.88 2.88 0 0 0 5.76 0z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
